### PR TITLE
(Improvement) Swap out ActionAPI regex paths for explicit paths

### PR DESCRIPTION
**Pull request by stevelacey**
This makes it so django will miss the route entirely if the action name is invalid, and improves output on the debug 404:

Screen Shot 2023-02-15 at 15 26 22

This also addresses a bug in that invalid action names currently serve a 405 for invalid action paths, and really those should be 404, but throwing a 404 from worf is not as good as making the django router miss and serve the router debug.

Also introduces an experimental `api` helper that adds sugar on top of `path` for defining api routes, meaning that ActionAPI views don't require an additional `path` reference in `urls.py`:

**Before:**

```py
path("interviews/", include([
    path("", staff.InterviewList.as_view()),
    path("<str:interview_id>/", include([
        path("", staff.InterviewDetail.as_view()),
        path("<str:action>/", staff.InterviewDetail.as_view()),
    ])),
])),
```

**After:**

```py
api("interviews/", staff.InterviewList, [
    api("<str:interview_id>/", staff.InterviewDetail),
]),
```

**TL;DR:** Actions are automatically read off of the view class and extra paths are registered for the same view, passing `action` in as an init kwarg; calling `.as_view()` is optional, if a class is supplied it will be called internally. Action paths always call it internally to pass in the `action` init kwarg.

I am still thinking about how much of a great idea this is, I've been reluctant to add this before, given urlpattern syntax is verbose because there's a lot of possibilities to cover, but ending the duplication of view references is appealing, as is registering all action paths explicitly.

This could be named `path`, and work as a drop in replacement for `django.urls` path, but that could be more confusing.

### CODE REVIEW

**Comment by stevelacey** [comment]
I may revive this one day but I don't much like replacing `path` with something else and it'd be better if the two things were interoperable. Perhaps the actions behavior should also be opt-in via a kwarg like `actions=True`.
